src/main/java/com/hubspot/ringleader/watcher/WatcherFactory.java

```diff
@@ -161,7 +186,15 @@ public void stateChanged(CuratorFramework client, ConnectionState newState) {
             case SUSPENDED:
             case LOST:
               LOG.info("Connection lost or suspended, replacing client");
-              replaceCurator(null);
+              replaceCurator();
+              break;
+            case CONNECTED:
+            case RECONNECTED:
+              for (PersistentWatcher watcher : watchers.keySet()) {
```

Comment:
Instead of the watchers passing runnables to ensure fetch is called again, the factory does it when it knows there's a new curator.